TestVeeamGZipStream/ParamsReader.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.IO;
+using TestVeeamGZipStream.Settings;
+using TestVeeamGZipStream.Settings.Mode;
+
+namespace TestVeeamGZipStream
+{
+    public class ParamsReader
+    {
+        public CompressionParams Read(string[] args)
+        {
+            if (args.Length != 3)
+            {
+                throw new ArgumentOutOfRangeException("Число параметров не соответствует ТЗ");
+            }
+            Mode mode = GetMode(args[0]);
```

Comment:
гетеры обычно не принимают параметры. назови лучше validate и не возвращай ничего а в конце создай объект 
CompressionParams
```suggestion
         ValidateMode(arg[0]);
         ValidateSourceFilePath(args[1]);
         ValidateOutputFilePath(args[2])
         
         return new CompressionParams(mode, sourceFile, recoverFileName);
            Mode mode = GetMode(args[0]);
```